### DISCUSSION TITLE
v0.0.9

### DISCUSSION THREAD

**Discussion by carson-katri** [carson-katri/dream-textures#420] [Announcements]
This update includes [texture projection](https://github.com/carson-katri/dream-textures/blob/main/docs/TEXTURE_PROJECTION.md), [inpainting model support](https://github.com/carson-katri/dream-textures/blob/main/docs/INPAINT_OUTPAINT.md), depth to image for the [render pass](https://github.com/carson-katri/dream-textures/blob/main/docs/RENDER_PASS.md), and a lot more. The internals have also been reimplemented to use [🤗 Diffusers](https://github.com/huggingface/diffusers).

Join the [Dream Textures Discord server](https://discord.gg/EmDJ8CaWZ7) to discuss the add-on, share your art, and get help from other users.

## Choose Your Installation
Several versions are available. Find the one that will work for you.

<details>
<summary>Windows</summary>

### NVIDIA GPU
* [Dream Textures for Windows \(CUDA\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-windows-cuda.7z) - Local and cloud rendering with NVIDIA CUDA
    * Due to file size limits on GitHub, the add-on ZIP file is contained in a 7-Zip archive.
    * Install [7-Zip](https://www.7-zip.org/), right click on the downloaded `dream_textures-windows-cuda.7z` archive, select "7-Zip" > "Extract Here" to get the `dream_textures-windows-cuda.zip` file.
    * In Blender preferences open "Add-ons" > "Install...", then choose the `dream_textures-windows-cuda.zip` file.
* [Dream Textures for DreamStudio \(Windows\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-windows-dreamstudio.zip) - Cloud rendering with DreamStudio

### AMD GPU
* [Dream Textures for DreamStudio \(Windows\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-windows-dreamstudio.zip) - Cloud rendering with DreamStudio
> *Note* Local generation is not available for AMD GPUs.
</details>

<details>
<summary>macOS</summary>

### Apple Silicon (M1/M2)
* [Dream Textures for Apple Silicon](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-macos-arm.zip) - Local and cloud rendering for Apple Silicon Macs
* [Dream Textures for DreamStudio \(Apple Silicon\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-macos-arm-dreamstudio.zip) - Cloud rendering with DreamStudio

### Intel
* [Dream Textures for DreamStudio \(Intel\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-macos-intel-dreamstudio.zip) - Cloud rendering with DreamStudio
> *Note* Local generation is not available for Intel Macs.
</details>

<details>
<summary>Linux</summary>

Installation must be completed manually. Follow the instructions under [Contributing](https://github.com/carson-katri/dream-textures/#contributing) for more details.
</details>

<details>
<summary>
<a href="https://www.blendermarket.com/products/dream-textures">Blender Market</a>
</summary>

You can optionally purchase it for a small fee on Blender Market.

> The version on Blender Market is identical to the free download available here. It is simply a way to support my work on this tool.

</details>

## Setup
After installing the appropriate add-on ZIP file, enable the add-on and expand it's preferences. There you will find further instructions for setup.

See the [setup guide](https://github.com/carson-katri/dream-textures/blob/main/docs/SETUP.md) for more detailed instructions.

### Texture Projection
Texture an entire scene at once using the depth to image model.

![](https://github.com/carson-katri/dream-textures/blob/main/docs/assets/texture_projection/projection.gif)

## Full Changelog
* Release v0.0.8 by @carson-katri in https://github.com/carson-katri/dream-textures/pull/359
* Add link to video tutorial by @carson-katri in https://github.com/carson-katri/dream-textures/pull/375
* Linux Fixes by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/384
* Show the seed in history to easier match generated images to their promts by @GottfriedHofmann in https://github.com/carson-katri/dream-textures/pull/396
* UI enhancement: When recalling a promt, open the corresponding image … by @GottfriedHofmann in https://github.com/carson-katri/dream-textures/pull/403
* 🤗 Diffusers Backend by @carson-katri in https://github.com/carson-katri/dream-textures/pull/405
* Outpainting for #405 by @carson-katri in https://github.com/carson-katri/dream-textures/pull/412
* Checkpoint File Import by @carson-katri in https://github.com/carson-katri/dream-textures/pull/413
* Update preferences.py by @eltociear in https://github.com/carson-katri/dream-textures/pull/416
* Stability SDK for #405 by @carson-katri in https://github.com/carson-katri/dream-textures/pull/411
* Add Project Dream Texture operator by @carson-katri in https://github.com/carson-katri/dream-textures/pull/409
* v0.0.9 documentation updates by @carson-katri in https://github.com/carson-katri/dream-textures/pull/417
* Release v0.0.9 by @carson-katri in https://github.com/carson-katri/dream-textures/pull/418

## New Contributors
* @GottfriedHofmann made their first contribution in https://github.com/carson-katri/dream-textures/pull/396
* @eltociear made their first contribution in https://github.com/carson-katri/dream-textures/pull/416

**Full Changelog**: https://github.com/carson-katri/dream-textures/compare/0.0.8...0.0.9

<hr /><em>This discussion was created from the release <a href='https://github.com/carson-katri/dream-textures/releases/tag/0.0.9'>v0.0.9</a>.</em>

**Comment by JC-Sutter** [comment]
Cool 🙂
So I think I must uninstall the v008 first. Is there anything else I must download? The .ckpt files for SD2.1 for example? Where do I find them?

**Reply by carson-katri** [reply]
No need to use ckpt files anymore (unless you have some you want to import). The add-on preferences has a built-in model search and download feature that lets you download any model from the Hugging Face Hub right inside Blender.

**Reply by JC-Sutter** [reply]
Wow cool. 🙂. But It's needed to unsinstall the v.008. Or can I just install the v009 parallel?

**Reply by carson-katri** [reply]
You need to uninstall 0.0.8

**Comment by Darshd994** [comment]
I can't download stable diffusion v2.1 (recommended). whenever I click on it it just shows the downloading bar and that's it.. it just always asks me to download again and again.

**Reply by JC-Sutter** [reply]
Yes, It's the same to me. I can't download stable diffusion v2.1 (recommended). I can't download any models, also not the ones in the search list.

**Reply by Darshd994** [reply]
Yep Same for me. not even the ones in the search list.

**Reply by wedgeewoo** [reply]
I'm getting the same thing, I wonder where the model directory is?

**Reply by JC-Sutter** [reply]
It seems not possible to import the already downloaded models from the Harddisk like in v008?

**Reply by wedgeewoo** [reply]
yeah I can't import any models or download any other models, weird.

**Reply by wedgeewoo** [reply]
ive tried reinstalling on the default directory for blender, and i still can't import or download any models

**Reply by wedgeewoo** [reply]
so heres what i did, I went to C:\Users\User\.cache\huggingface\diffusers and deleted all the diffuser folders in there, and then went back to blender and reinstalled SD2.1. It said it was an unsupported model so I downloaded SD2.0 Depth and it seems to work now. Ive tried importing ckpt files from my own sources but they don't work like the other directories . Anyway this is a really cool application, just have to play around a bit,

**Reply by carson-katri** [reply]
When importing checkpoints make sure you select the model type in the sidebar. You can choose from v1, v2 (512), and v2 (768) options. This should work to import dreambooth models not on huggingface

**Reply by JC-Sutter** [reply]
When I try to import the Checkpoint File "v1-5-purned.ckpt"  I get an error

![Clipboarder 2022 12 16](https://user-images.githubusercontent.com/76898123/208123380-2fe3cd06-2c2d-4ea6-94fa-ab53e6121ed7.png)
![image](https://user-images.githubusercontent.com/76898123/208123427-86efd04d-c5e4-4a5f-b48a-17634ed04d58.png)

**Reply by wedgeewoo** [reply]
oh right on it works. I assume for the dream projection it will only use the depth model.

**Reply by JC-Sutter** [reply]
I don't understand what's the issue? I'm new with this tools.
In C:\Users\User.cache\huggingface\ I have no diffusers folder. What exactly do I need to install and where can I downdload it, if the autmatic download is not working? What is ment with reinstalling SD2.0 to Blender? In v008 I did not need to install SD2.0?

**Reply by carson-katri** [reply]
Do you have any errors appearing in Window > Toggle System Console?

**Reply by JC-Sutter** [reply]
This is the Toggle System Console after installing and trying to download the models and import a ckpt modul


Read prefs: C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\config\userpref.blend
Reloading external rigs...
Reloading external metarigs...
3.03
Files Should be Exporting to : C:\Users\User\Documents\DAZ 3D\Bridges\Daz To Blender\Exports

Loading DAZ
DAZ loaded
Load settings from C:/Users/User/import-daz-settings-28x.json
botaniq install dir autodetected as C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\botaniq_full (big zip embedded)
Update Scrollbars: ['Cube', 'Light', 'Camera']
addon_utils.disable: dream_textures not disabled
Modules Installed (dream_textures) from 'D:\\downloads\\Dreamtexture009\\dream_textures-windows-cuda.zip' into 'C:\\Users\\User\\AppData\\Roaming\\Blender Foundation\\Blender\\3.3\\scripts\\addons'
Exception: Traceback (most recent call last):
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\generator_process\actor.py", line 271, in _receive
    response = getattr(self, message.method_name)(*message.args, **message.kwargs)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\generator_process\actions\huggingface_hub.py", line 62, in hf_list_installed_models
    from diffusers.utils import DIFFUSERS_CACHE
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\diffusers\__init__.py", line 27, in <module>
    from .modeling_utils import ModelMixin
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\diffusers\modeling_utils.py", line 52, in <module>
    import accelerate
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\accelerate\__init__.py", line 7, in <module>
    from .accelerator import Accelerator
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\accelerate\accelerator.py", line 33, in <module>
    from .tracking import LOGGER_TYPE_TO_CLASS, GeneralTracker, filter_trackers
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\accelerate\tracking.py", line 45, in <module>
    import wandb
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\site-packages\wandb\__init__.py", line 26, in <module>
    from wandb import sdk as wandb_sdk
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\site-packages\wandb\sdk\__init__.py", line 9, in <module>
    from .wandb_init import _attach, init  # noqa: F401
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\site-packages\wandb\sdk\wandb_init.py", line 31, in <module>
    from .backend.backend import Backend
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\site-packages\wandb\sdk\backend\backend.py", line 19, in <module>
    from ..internal.internal import wandb_internal
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\site-packages\wandb\sdk\internal\internal.py", line 27, in <module>
    import psutil
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\psutil\__init__.py", line 109, in <module>
    from . import _pswindows as _psplatform
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\.python_dependencies\psutil\_pswindows.py", line 35, in <module>
    from ._psutil_windows import ABOVE_NORMAL_PRIORITY_CLASS
ImportError: DLL load failed while importing _psutil_windows: Das angegebene Modul wurde nicht gefunden.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\generator_process\actor.py", line 111, in run_callbacks
    done_callback(self)
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\preferences.py", line 185, in on_done
    set_model_list('installed_models', future.result())
  File "C:\Users\User\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\generator_process\actor.py", line 58, in result
    raise self._exception
ImportError: DLL load failed while importing _psutil_windows: Das angegebene Modul wurde nicht gefunden.

**Reply by JC-Sutter** [reply]
The only Errors I find in the Console are

ImportError: DLL load failed while importing _psutil_windows: Das angegebene Modul wurde nicht gefunden.
mportError: DLL load failed while importing _psutil_windows: Das angegebene Modul wurde nicht gefunden.

No chance to get it run?

**Reply by JC-Sutter** [reply]
I have installed Blender 3.4. There v009 is runing now 🙂
So now I can run v008 on Blender 3.3 and v009 on Blender 3.4. I need that, because I can't load the .json setting files from v008 in v009. I get the error v1-5-pruned-emaonly not found, but I have import that model and it's also working when I transver the prompts manualy

**Reply by JC-Sutter** [reply]
... but it's  not possible to expport or import prompts.json. It get errors:  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\python\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '

and I also get strange results in the v2 models when I compare between Dreamtexture and Automatic1111. But the v1 Models are working well


Left ist DreamTexture and right is automatic1111 with the same settings
![image](https://user-images.githubusercontent.com/76898123/208233877-a883862a-9849-43dd-80aa-90631592846d.png)

**Reply by gordeyr** [reply]
> Я установил Блендер 3.4. Там v009 сейчас работает🙂 Так что теперь я могу запустить v008 на Blender 3.3 и v009 на Blender 3.4. Мне это нужно, потому что я не могу загрузить файлы настроек .json из v008 в v009. Я получаю сообщение об ошибке v1-5-pruned-emaonly not found, но я импортировал эту модель, и она также работает, когда я выполняю подсказки вручную.

I still have the problem on blender version 3.4.1

**Comment by osvoldon** [comment]
Is there a way to use the old esrgan upscalling? The new method is great, but sometimes can be impractical due to longer processing time.

**Comment by JC-Sutter** [comment]
Is there a way to dowload and import models wich are listed in the Hugging Face Hub to import and use in v008? v009 is not running correct on my PC but I would like to work with the "kingery/hyc-01-sdv1-5-2e-6-1500-man-ddim" model in v008.

**Reply by carson-katri** [reply]
You need a `ckpt` file for v0.0.8, like automatic1111 uses

**Comment by ProStoKrut123** [comment]
Help, please, when generating textures, instead of the expected result with any texture, it produces a black texture. What could be the problem?

**Reply by carson-katri** [reply]
Do you have NVIDIA 1650/1660? If so, under advanced turn off “Half Precision”

**Reply by ProStoKrut123** [reply]
Yes, NVIDIA 1660 ti to be exact. I did everything as you said, but the generation stopped at the "Starting ..." stage.

**Reply by ProStoKrut123** [reply]
Console output:
You have disabled the safety checker for <class 'dream_textures.generator_process.actions.depth_to_image.depth_to_image.<locals>.GeneratorPipeline'> by passing `safety_checker=None`. Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results. For more information, please have a look at https://github.com/huggingface/diffusers/pull/254 .
  0%|                                                                                           | 0/25 [00:03<?, ?it/s]
Exception in thread Thread-2 (_send_thread):
Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Vlad\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\generator_process\actor.py", line 299, in _send_thread
    response = self._response_queue.get()
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\python\lib\multiprocessing\queues.py", line 122, in get
    return _ForkingPickler.loads(res)
ModuleNotFoundError: No module named 'torch'

**Comment by Srcko7** [comment]
Hello,

i have the same issue, after generation i get a black picture i have a Qwadro T1000 Qmax with 4GB, id love to give this a try, how could i get it ruining?

**Comment by UrsulaMedium** [comment]
Hey Hey! It seems the DreamStudio version is not able to use a model with Depth features yet. Or am I doing something wrong?
<img width="292" alt="image" src="https://user-images.githubusercontent.com/35008564/209132592-3915bca5-2981-483a-aae0-ef57b9596560.png">

**Reply by L-Travieso** [reply]
interesting blender theme, can you share it?

**Comment by Soapheart** [comment]
Hello there!
Is it possible to change a work folder for models?
Default is "C:\Users\%username%\.cache\huggingface\diffusers" and cause my profile folder includes some cyrillic characters - addon cant work properly with it.

**Reply by NullSenseStudio** [reply]
This should be possible by setting your HF_HOME system environment variable to where you want the huggingface folder to be.

**Reply by Soapheart** [reply]
Thanks, it works!
After setting system variable necessary to reboot your system, after that - changes will apply.
Happy holidays!

**Reply by Dextrale** [reply]
> This should be possible by setting your HF_HOME system environment variable to where you want the huggingface folder to be.

please tell me step by step how to start it and what to prescribe where?

**Comment by LDFM5** [comment]
When trying to download Stable Difussion v2.1 I get this error:

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\luisd\AppData\Roaming\Blender Foundation\Blender\3.4\scripts\addons\dream_textures\generator_process\actor.py", line 111, in run_callbacks
    done_callback(self)
  File "C:\Users\luisd\AppData\Roaming\Blender Foundation\Blender\3.4\scripts\addons\dream_textures\preferences.py", line 145, in on_done
    set_model_list('installed_models', Generator.shared().hf_list_installed_models().result())
  File "C:\Users\luisd\AppData\Roaming\Blender Foundation\Blender\3.4\scripts\addons\dream_textures\generator_process\actor.py", line 64, in result
    raise self._exception
NameError: name '_C' is not defined

**Comment by Rodrigotechnology** [comment]
I have this SyntaxError
Traceback (most recent call last):
  File "D:\Programas\3.0\scripts\modules\addon_utils.py", line 351, in enable
    mod = __import__(module_name)
  File "C:\Users\Usuario\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\dream_textures\__init__.py", line 39, in <module>
    from .render_pass import register_render_pass, unregister_render_pass, pass_inputs
  File "C:\Users\Usuario\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\dream_textures\render_pass.py", line 90
    match scene.dream_textures_render_properties_pass_inputs:
          ^
SyntaxError: invalid syntax

**Reply by carson-katri** [reply]
You need Blender 3.1 or higher

**Reply by Rodrigotechnology** [reply]
Tks.

**Reply by Rodrigotechnology** [reply]
How could fix if get a message Unsopported model? I´m using version 2.1

**Reply by JasonHoku** [reply]
> How could fix if get a message Unsopported model? I´m using version 2.1

I'm pretty sure that error shows if you are trying to use depth on a non-depth model. 

So probably add and select a model with depth capability OR for non depth models, use the Image Editor tab or the Cycles Render Dream Textures options and make sure to not select Depth in the Pass Input

**Comment by lepoche** [comment]
I don't know if anyone had problems using it but basically once I select the model and write the subject, nothing happens. It always keeps starting the process and it doesn't advance, it doesn't throw me any errors. I have Blender 3.4, and an NVIDIA 1650. I tried unchecking **Half processing**, but nothing happened.

**Reply by TomHardy02** [reply]
@lepoche Did you ever solve your issue, I am experiencing the same problem.

**Comment by L-Travieso** [comment]
Please bring back the "Delete Selected Weights" Button under the model list from v0.0.8 into v0.0.9.
I can't remove installed weights completely with the new Folder Icon method on the right side of the list in v0.0.9. The model entries remain in the list even if the folders have been removed. It's a mess the way it is right now.
How does one clear out the list entries of ghost (deleted) models?
